arch/arm/core/aarch64/fatal.c

```diff
@@ -222,3 +223,17 @@ void z_arm64_fatal_error(unsigned int reason, z_arch_esf_t *esf)
 
 	CODE_UNREACHABLE;
 }
+
+/**
+ * @brief Handle a software-generated fatal exception
+ * (e.g. kernel oops, panic, etc.).
+ *
+ * @param esf exception frame
+ */
+void z_arm64_do_kernel_oops(z_arch_esf_t *esf)
```

Comment:
I don't know if it's your choice but userspace part is missing.
```

 * If the caller is running in user mode, only K_ERR_KERNEL_OOPS or
 * K_ERR_STACK_CHK_FAIL may be induced.
```
Without this, lot of tests related to userspace will failed. Enjoy, it's already done for arch32

```
#if defined(CONFIG_USERSPACE)
	if ((__get_CONTROL() & CONTROL_nPRIV_Msk) == CONTROL_nPRIV_Msk) {
		/*
		 * Exception triggered from nPRIV mode.
		 *
		 * User mode is only allowed to induce oopses and stack check
		 * failures via software-triggered system fatal exceptions.
		 */
		if (!((esf->basic.r0 == K_ERR_KERNEL_OOPS) ||
			(esf->basic.r0 == K_ERR_STACK_CHK_FAIL))) {

			reason = K_ERR_KERNEL_OOPS;
		}
	}

#endif /* CONFIG_USERSPACE */
```

Of course it could be done later on another commit.

Reply:
I know it but userspace is not in the picture at the moment. But even without usermode this is needed to have actual oops-ses & company triggered in the correct way.